### ISSUE TITLE
Planned Upgrades to Debug Info Support

### ISSUE THREAD

**Issue by adinn** [oracle/graal#5047]
### TL;DR
The following tasks are outstanding:
 - Implement debug info generation for truffle-based images at runtime
 - Implement debug info generation on MacOS
 - Implement debug type and local var info generation on Windows
 - Improve Linux debug info support for Linux tools (perf and valgrind)

### Goals

- Enable high quality debugging on all supported OSes equivalent to that available on Linux/gdb as far as possible.
- Enable high quality analysis of code performance and memory usage via Linux tools that employ debug info

### Non-Goals

- Not currently expecting to support debug capabilities beyond those available on Linux/gdb.

**Comment by jerboaa** [comment]
> Implement debug type and local var info generation on Windows

PR for this is here? https://github.com/oracle/graal/pull/3732/

**Comment by adinn** [comment]
> > Implement debug type and local var info generation on Windows

> PR for this is here? https://github.com/oracle/graal/pull/3732

Yes, I believe that PR includes all that was planned by way of functionality. It will probably also need some doc updates.

> - Improve Linux debug info support for Linux tools (perf and valgrind) 

n.b. The PR which completes this task is #4950

**Comment by stooke** [comment]
> > Implement debug type and local var info generation on Windows
> 
> PR for this is here? #3732

#3732 does not include local var support.  That's coming up next.
